From 88d9e1b2625519e612219bb9f1339b06f46b0f2b Mon Sep 17 00:00:00 2001
From: linmin <linmin@eswincomputing.com>
Date: Wed, 22 May 2024 16:21:19 +0800
Subject: [PATCH 049/109] WIN2030-15099:fix(console webserver):Get power status
 before reboot operation

Changelogs:
1.Modified console.c, reboot is only available when som power is on
2.Modified web-server.c, get_power_info() only when som power is on

Change-Id: I046f48fe18954e566c856944ec1bdd70134a1a69
---
 Core/Src/console.c          | 21 +++++++++++++--------
 Core/Src/hf_power_process.c |  3 +++
 Core/Src/web-server.c       | 20 +++++++++++---------
 3 files changed, 27 insertions(+), 17 deletions(-)

diff --git a/Core/Src/console.c b/Core/Src/console.c
index 9230aaf..a8ef1d8 100644
--- a/Core/Src/console.c
+++ b/Core/Src/console.c
@@ -1100,15 +1100,20 @@ static BaseType_t prvCommandReboot(char *pcWriteBuffer, size_t xWriteBufferLen,
 {
     int ret = HAL_OK;
 
-    ret = web_cmd_handle(CMD_RESET, NULL, 0, 1000);
-    if (HAL_OK != ret) {
-        som_reset_control(pdTRUE);
-        osDelay(10);
-        som_reset_control(pdFALSE);
-        printf("Faild to reboot SOM(ret %d), force reset SOM!\n", ret);
+    if (SOM_POWER_ON == get_som_power_state()) {
+        ret = web_cmd_handle(CMD_RESET, NULL, 0, 1000);
+        if (HAL_OK != ret) {
+            som_reset_control(pdTRUE);
+            osDelay(10);
+            som_reset_control(pdFALSE);
+            printf("Faild to reboot SOM(ret %d), force reset SOM!\n", ret);
+        }
+        // Trigger the Som timer to enusre SOM could reboot in 5 senconds
+        TriggerSomRebootTimer();
+    }
+    else {
+         printf("Err, som board is in power off status, can NOT be reboot!!!\n");
     }
-    // Trigger the Som timer to enusre SOM could reboot in 5 senconds
-    TriggerSomRebootTimer();
 
     return pdFALSE;
 }
diff --git a/Core/Src/hf_power_process.c b/Core/Src/hf_power_process.c
index 3f7991e..1705953 100644
--- a/Core/Src/hf_power_process.c
+++ b/Core/Src/hf_power_process.c
@@ -169,6 +169,9 @@ void hf_power_task(void *parameter)
 	osDelay(1000);
 	if(es_autoboot())
 		change_som_power_state(SOM_POWER_ON);
+	else
+		change_som_power_state(SOM_POWER_OFF);
+
 #ifdef AUTO_BOOT
 	change_som_power_state(SOM_POWER_ON);
 #endif
diff --git a/Core/Src/web-server.c b/Core/Src/web-server.c
index 2f4a8b7..3578319 100644
--- a/Core/Src/web-server.c
+++ b/Core/Src/web-server.c
@@ -1,4 +1,5 @@
 
+#include <stdlib.h>
 #include "lwip/opt.h"
 #include "lwip/arch.h"
 #include "lwip/api.h"
@@ -32,12 +33,12 @@
 /* define ------------------------------------------------------------*/
 #define WEB_DEBUG_EN	0
 #define web_fmt(fmt)	"[%s-WEB]: " fmt
-#define dbg_web_fmt(fmt)	web_fmt("%s[%d]: " fmt), "DEBUG",	\
+#define web_dbg_fmt(fmt)	web_fmt("%s[%d]: " fmt), "DEBUG",	\
 		        __func__, __LINE__
 #if WEB_DEBUG_EN
 #define web_debug(fmt, args...) \
 	do {							\
-		printf(dbg_web_fmt(fmt), ##args);	\
+		printf(web_dbg_fmt(fmt), ##args);	\
 	} while (0)
 #else
 #define web_debug(fmt, args...)
@@ -9997,7 +9998,7 @@ char* concatenate_strings(const char* str1, const char* str2) {
     int length = strlen(str1) + strlen(str2) + 1; // +1 for the null terminator
     char* new_str = malloc(length);
     if (new_str == NULL) {
-        web_debug(stderr, "Memory allocation failed\n");
+        web_debug("Memory allocation failed\n");
         exit(1);
     }
 
@@ -10413,12 +10414,12 @@ typedef struct  {
 POWERInfo get_power_info(void){
 
 
-	POWERInfo example = {
-        60,5,12
-    };
-	get_board_power(&example.voltage,&example.current,&example.consumption);
-	//get_som_power(&example.voltage,&example.current,&example.consumption);
-    return example;
+	POWERInfo example = {0};
+
+	if (get_power_status())
+		get_board_power(&example.voltage,&example.current,&example.consumption);
+
+	return example;
 }
 
 
@@ -11665,6 +11666,7 @@ int get_soc_status()
 			netconn_close(newconn);
 			netconn_delete(newconn);
 		}
+		// osDelay(1000);
 	}
 	while (1);
 	netconn_close(conn);
-- 
2.25.1

